InteractionBase/src/commands/CAddNodeToViewByName.cpp

```diff
@@ -84,8 +94,14 @@ QList<CommandSuggestion*> CAddNodeToViewByName::suggest(Visualization::Item*, Vi
 		auto nodeName = textSoFar.trimmed().mid(4);
 		QStringList names;
 		for (auto manager : Model::AllTreeManagers::instance().loadedManagers())
-			names.append(findNames(nodeName.split("."), manager->root()));
+		{
+			auto parts = nodeName.split(".");
+			QString pattern = "*";
+			for (auto part : parts) pattern += part + '*';
 
+			names.append(findNames(new Model::SymbolMatcher(new QRegExp(pattern, Qt::CaseInsensitive, QRegExp::Wildcard)),
```

Comment:
There are a few things wrong with this loop:
* The first three lines are identical in all iterations, please put them before the loop
* The part `new Model::SymbolMatcher(new QRegExp(pattern, Qt::CaseInsensitive, QRegExp::Wildcard))` is also identical and should go before the loop
* That same part from point 2 introduces a memory leak. You create many Symbol matchers, but they are not deleted anywhere. The correct thing to do is, create a single matcher, before the loop, without using `new`. The modify your findNames method to take a `const SymbolMatcher& matcher` argument instead of a pointer and just pass the object from before. This way, the matcher will be automatically remove once it's out of scope. (don't worry about the `new` of QRegExp. The matcher takes ownership of that object and will delete it.